substrate/network-libp2p/src/network_state.rs

```diff
@@ -175,17 +176,35 @@ impl NetworkState {
 		let node_store = if let Some(ref path) = config.net_config_path {
 			let path = Path::new(path).join(NODES_FILE);
 			if let Ok(node_store) = JsonPeerstore::new(path.clone()) {
-				debug!(target: "sub-libp2p", "Initialized peer store for JSON \
-					file {:?}", path);
+				debug!(target: "sub-libp2p", "Initialized peer store for JSON file {:?}", path);
 				NodeStore::Json(node_store)
 			} else {
-				warn!(target: "sub-libp2p", "Failed to open peer storage {:?} \
-					; peers won't be saved", path);
-				NodeStore::Memory(MemoryPeerstore::empty())
+				warn!(target: "sub-libp2p", "Failed to open peer storage {:?}; peers file will be reset", path);
+				fs::remove_file(&path).expect("Failed deleting peers.json");
+
+				// we check for about 1s if the file was really deleted and move on
+				for _x in 0..200 {
+					if !Path::new(&path).exists() {
+						break;
+					} else {
+						debug!("Waiting for effective deletion of invalid/outdate peers.json");
+						thread::sleep(time::Duration::from_millis(5));
+					}
+				}
+
+				if let Ok(peerstore) = JsonPeerstore::new(path.clone()) {
+					debug!("peers.json reset");
+					NodeStore::Json(peerstore)
+				} else {
+					warn!(target: "sub-libp2p",
+						"Failed to reset peer storage {:?}; peers change will not be saved",
```

Comment:
This can also happen if the user doesn't have the permission to open the file, so the message should just be `Failed to open peer storage` IMO.

Reply:
Would not the message L182 already say that?
I can change the message, no problem. We would then have 2 identical messages. Would you prefer that?

If the file cannot be open because of permissions, this should be caught by the first tests. The message `Failed to reset peer storage` really comes second, only after we tried deleting and recreating the file.